slither/detectors/abstract_detector.py

```diff
@@ -139,6 +144,12 @@ def _log(self, info: str) -> None:
         if self.logger:
             self.logger.info(self.color(info))
 
+    def _uses_vulnerable_solc_version(self) -> bool:
+        if not self.VULNERABLE_SOLC_VERSIONS:
+            return False
+
+        return self.compilation_unit.solc_version in self.VULNERABLE_SOLC_VERSIONS
```

Comment:
```suggestion
    def _uses_vulnerable_solc_version(self) -> bool:
        if self.VULNERABLE_SOLC_VERSIONS:
            return self.compilation_unit.solc_version in self.VULNERABLE_SOLC_VERSIONS

        return True
```
The base case should be true so that the detector runs on all versions unless specified in the detector implementation.

Reply:
oops, no wonder all my tests are failing. thank you 🙏🏽